apps/meteor/client/hooks/roomActions/useE2EERoomAction.ts

```diff
@@ -26,20 +30,53 @@ export const useE2EERoomAction = () => {
 	const { t } = useTranslation();
 	const { otrState } = useOTR();
 
+	const enabledOnRoom = !!room.encrypted;
+
+	const roomType = getRoomTypeTranslation(room).toLowerCase();
+	const roomId = room._id;
+
 	const toggleE2E = useEndpoint('POST', '/v1/rooms.saveRoomSettings');
 
+	const canResetRoomKey = enabled && isE2EEReady && (room.t === 'd' || permittedToToggleEncryption);
+
 	const action = useEffectEvent(async () => {
 		if (otrState === OtrRoomState.ESTABLISHED || otrState === OtrRoomState.ESTABLISHING || otrState === OtrRoomState.REQUESTED) {
 			dispatchToastMessage({ type: 'error', message: t('E2EE_not_available_OTR') });
 
 			return;
 		}
 
+		if (enabledOnRoom) {
+			imperativeModal.open({
+				component: BaseDisableE2EEModal,
+				props: {
+					onClose: imperativeModal.close,
+					onConfirm: handleToogleE2E,
+					roomType,
+					roomId,
+					canResetRoomKey,
+				},
+			});
+		} else {
+			imperativeModal.open({
+				component: EnableE2EEModal,
+				props: {
+					onClose: imperativeModal.close,
+					onConfirm: handleToogleE2E,
+					roomType,
+				},
+			});
+		}
+	});
```

Comment:
Couldn't you have used `setModal` here?

Reply:
I think I could if I change the file to use `tsx`, but I followed what the other room actions are doing...